templates/commands/render_action_append.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+
+	"github.com/abcxyz/abc/templates/model"
+)
+
+func actionAppend(ctx context.Context, ap *model.Append, sp *stepParams) error {
+	path, err := parseAndExecuteGoTmpl(ap.Path.Pos, ap.Path.Val, sp.inputs)
+	if err != nil {
+		return err
+	}
+
+	with, err := parseAndExecuteGoTmpl(ap.With.Pos, ap.With.Val, sp.inputs)
+	if err != nil {
+		return err
+	}
+
+	if err := walkAndModify(ctx, ap.Path.Pos, sp.fs, sp.scratchDir, path, func(buf []byte) ([]byte, error) {
+		// todo: should we add a newline before/after appending or leave that to user?
```

Comment:
Discussion question #1

Reply:
My gut feeling is "no". It's fairly easy to communicate "the string is appended exactly as-is; no newlines are added" versus adding the newline and then needing to support a use case where the customer doesn't want to add the newline (e.g. they want to add a trailing line comment).

Reply:
(inlining for easier reading)
@sethvargo  My gut feeling is "no". It's fairly easy to communicate "the string is appended exactly as-is; no newlines are added" versus adding the newline and then needing to support a use case where the customer doesn't want to add the newline (e.g. they want to add a trailing line comment).

Reply:
I guess the main thing I'm not sure how to fix would be if someone wants to remove the trailing newlines before appending. Worth making an extra flag or something?

```
This is some text file.
It has text.
I want to finish this
```

If you wanted to append `sentence.` to the end, currently you would be forced to do it after the newline `append("sentence\n)`

```
This is some text file.
It has text.
I want to finish this
 sentence.
```

What you really want is
```
This is some text file.
It has text.
I want to finish this sentence.
```

I guess you might be able to do it with a regex replace though, as it seems that the default is not to match "end of text" line-per-line but with the entire file https://github.com/google/re2/wiki/Syntax

^^ Might be worth considering if we should document that in regex_replace, as its very different behavior than most people expect with regex from GREP.

Reply:
Added regex replace documentation: #118

Reply:
Hmm, if we do that, then this we would need a special flag for newline, as there wouldn't be another way to append to the end of a file but remove the \n

Reply:
What seems best to me is to ensure the file ends with newline, by default. The user could disable this with a new option `ensure_ending_newline: false`.

Reply:
I didn't read closely enough, you are suggesting a special mode for regex spanning lines.

Reply:
Added new param "skip-ensure-newline" and updated documentaiton. I am wondering if I need to support \r\n newlines or not

Reply:
I think we shouldn't support \r\n newlines at, least for now.